pkg/query-service/model/response.go

```diff
@@ -269,6 +269,32 @@ type SearchSpanResponseItem struct {
 	SpanKind         string            `json:"spanKind"`
 }
 
+type SearchSpanResponseItemV2 struct {
+	TimeUnixNano      time.Time          `json:"timestamp" ch:"timestamp"`
+	DurationNano      uint64             `json:"durationNano" ch:"duration_nano"`
+	SpanID            string             `json:"spanId" ch:"span_id"`
+	TraceID           string             `json:"traceId" ch:"trace_id"`
+	HasError          bool               `json:"hasError" ch:"has_error"`
+	Kind              int8               `json:"kind" ch:"kind"`
+	ServiceName       string             `json:"serviceName" ch:"resource_string_service$$name"`
+	Name              string             `json:"name" ch:"name"`
+	References        string             `json:"references,omitempty" ch:"references"`
+	Attributes_string map[string]string  `json:"attributes_string" ch:"attributes_string"`
+	Attributes_number map[string]float64 `json:"attributes_number" ch:"attributes_number"`
+	Attributes_bool   map[string]bool    `json:"attributes_bool" ch:"attributes_bool"`
+	Events            []string           `json:"event" ch:"events"`
```

Comment:
Same as above. The event should be easily consumed without having to client (API or frontend to json deserialize it). It may have been string till now but it doesn't make sense to maintain status quo with new traces work.

Reply:
Span event has a strictly typed structure in the protocol. What is the reason for returning a JSON string in the response?

Reply:
To keep the API and FE working as it as. Also you mentioned that span evens are a seperate piece in itself which itself need another set of work

https://github.com/SigNoz/signoz/issues/2767

Reply:
The ability to query span events has no relation with how they are sent in response for trace detail

Reply:
Sorry for the confusion, it's not the actual response item, I have removed the json tags and moved it to a separete file. Thanks for pointing out.